common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/responses/signin/SignInInitiateApiResponse.kt

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.providers.nativeauth.responses.signin
+
+import com.google.gson.annotations.Expose
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.providers.nativeauth.IApiResponse
+import com.microsoft.identity.common.java.providers.nativeauth.interactors.InnerError
+import com.microsoft.identity.common.java.util.isInvalidGrant
+import com.microsoft.identity.common.java.util.isRedirect
+import com.microsoft.identity.common.java.util.isUserNotFound
+import java.net.HttpURLConnection
+
+/**
+ * Represents the raw response from the /initiate endpoint.
+ * Can be converted to SignInInitiateApiResult using the provided toResult() method.
+ */
+data class SignInInitiateApiResponse(
+    @Expose override var statusCode: Int,
+    @Expose @SerializedName("credential_token") val credentialToken: String?,
```

Comment:
Were we renaming this to flow token? @SammyO

Reply:
Flow token is the general term, but the API returns it as credential_token, signup_token and password_reset_token. So this code here is correct.
note: this will most likely be changed in the next iteration of the API, creating one parameter name for all tokens across all APIs.

Reply:
Sounds good